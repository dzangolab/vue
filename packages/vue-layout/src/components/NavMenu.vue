<template>
  <nav class="nav-menu">
    <div v-if="sideBarMenuToShow?.length > 0" class="menu">
      <NavMenuItem
        v-for="(menuItem, index) in sideBarMenuToShow"
        :key="index"
        :class="{
          'active-item': activeIndex === index,
          active: activeIndex === index && !menuItem.children?.length,
        }"
        :item="menuItem"
        :sidebar-active="sidebarActive"
        @click="activeIndex = index"
      />
    </div>
  </nav>
</template>

<script lang="ts">
export default {
  name: "NavMenu",
};
</script>

<script setup lang="ts">
import { ref, computed } from "vue";

import NavMenuItem from "./NavMenuItem.vue";

import type { SidebarMenu } from "../types";
import type { PropType } from "vue";

const props = defineProps({
  menu: {
    required: true,
    type: Array as PropType<SidebarMenu[]>,
  },
  sidebarActive: {
    default: true,
    required: true,
    type: Boolean,
  },
});

const activeIndex = ref<number | null>(null);

const sideBarMenuToShow = computed(() => {
  return props.menu?.filter((menu) => !menu.hide);
});
</script>

<style lang="css">
.nav-menu > .menu {
  --_border-color: var(--border-color, #dbdbdb);
  --_padding-h: var(--padding-h, 0);
  --_padding-v: var(--padding-v, 1.5rem);

  border-bottom: 1px solid var(--_border-color);
  padding-bottom: var(--_padding-v);
  padding-left: var(--_padding-h);
  padding-top: var(--_padding-v);
}

.nav-menu > .menu > .active {
  --_bg-color: var(--nav-menu-bg-color, #0870e5);

  background-color: var(--_bg-color);
}
</style>
