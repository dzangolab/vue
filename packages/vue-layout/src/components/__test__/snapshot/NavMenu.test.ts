import { shallowMount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";

import type { VueWrapper } from "@vue/test-utils";

import NavMenu from "@/components/NavMenu.vue";

describe("NavMenu", () => {
  const wrapper: VueWrapper = shallowMount(NavMenu, {
    props: {
      menu: [
        {
          name: "Menu1",
          routeName: "menu-1",
          shortName: "M1",
        },
        {
          name: "Menu2",
          routeName: "menu-2",
          shortName: "M2",
        },
      ],
      sidebarActive: true,
    },
  });

  it("matches snapshot ", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
