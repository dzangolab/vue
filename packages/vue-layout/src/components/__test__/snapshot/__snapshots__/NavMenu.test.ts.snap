// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavMenu > matches snapshot  1`] = `
<nav
  class="nav-menu"
>
  <div
    class="menu"
  >
    
    <nav-menu-item-stub
      item="[object Object]"
      sidebaractive="true"
    />
    <nav-menu-item-stub
      item="[object Object]"
      sidebaractive="true"
    />
    
  </div>
</nav>
`;
